core/src/main/java/org/opensearch/sql/datasource/DataSourceServiceImpl.java

```diff
@@ -103,18 +112,58 @@ private void validateDataSourceMetaData(DataSourceMetadata metadata) {
     Preconditions.checkArgument(
         !Strings.isNullOrEmpty(metadata.getName()),
         "Missing Name Field from a DataSource. Name is a required parameter.");
-    Preconditions.checkArgument(
```

Comment:
why remove the duplication check? cover in other place?

Reply:
Duplicates are covered while saving the object in opensearch. Also dataSourceMap might not contain the latest data if a datasource with similar name is created in another node.

Reply:
More validation checks will come in next PR.